src/main/scala/com/databricks/spark/avro/AvroSaver.scala

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2015 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.databricks.spark.avro
+
+import java.sql.Timestamp
+import java.util.HashMap
+
+import scala.collection.immutable.Map
+
+import org.apache.spark.sql._
+import org.apache.spark.SparkContext._
+
+import org.apache.avro.generic.GenericData.Record
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.mapred._
+import org.apache.avro.SchemaBuilder.{RecordBuilder, FieldAssembler}
+import org.apache.avro.{SchemaBuilder, Schema}
+
+import org.apache.hadoop.io.NullWritable
+import org.apache.hadoop.mapred.JobConf
+
+object AvroSaver {
+  /* This object provides a save() method that is used to save SchemaRDD as avro file at a
+   * particular destination. To do this, we first convert the schema of SchemaRDD to avro schema,
+   * and then call convert each row of the rdd to corresponding avro types. When doing schema
+   * convertion, we are passing the builder object around and adding stuff to it.
+   * Another remark worth mentioning is the structName parameter that functions have. Avro
+   * records have a name associated with them, and those names must be unique. Since SturctType
+   * in sparkSQL doesn't have a name associated with it, we are taking the name of the last
+   * structure field that the current structure belongs to (in sparkSQL fields have a name, just as
+   * they do in avro). For example if the row at the top level had a field called "Auto", which
+   * happens to be a structure describing a car, we would call that structure "Auto". Note that in
+   * * case of schema like the following: Structure("A":1, "B": Array(Structure("X":1))), the inner
+   * * structure will get a name "B". When we process original rows, they get a name
+   * "topLevelRecord" (since we can't apply the rule above to them).
+   */
+
+  private def convertTypeToAvro[T](dataType: DataType,
```

Comment:
nit: putting all arguments at four-space indentation on their own line is more typical than aligning with the method name. i.e.,

```scala
pricate def convertTypeToAvro[T](
    dataType: DataType,
    ...
    ...): T = {
  dataType match {
  }
}
```